gtsfm/configs/synthetic_front_end.yaml

```diff
@@ -0,0 +1,97 @@
+# Synthetic front-end configuration specifically for the Tanks & Temples dataset.
+
+SceneOptimizer:
+  _target_: gtsfm.scene_optimizer.SceneOptimizer
+  save_gtsfm_data: True
+  save_two_view_correspondences_viz: False
+  save_3d_viz: True
+  pose_angular_error_thresh: 5 # degrees
+
+  # retriever:
+  #   _target_: gtsfm.retriever.sequential_retriever.SequentialRetriever
+  #   max_frame_lookahead: 4
+
+  # retriever:
+  #   _target_: gtsfm.retriever.netvlad_retriever.NetVLADRetriever
+  #   num_matched: 50
+  #   min_score: 0.3
+
+  retriever:
+    _target_: gtsfm.retriever.joint_netvlad_sequential_retriever.JointNetVLADSequentialRetriever
+    num_matched: 2
+    min_score: 0.2
+    max_frame_lookahead: 3
+
+
+  correspondence_generator:
+    _target_: gtsfm.frontend.correspondence_generator.synthetic_correspondence_generator.SyntheticCorrespondenceGenerator
+    #dataset_root: /Users/johnlambert/Downloads/Tanks_and_Temples_Barn_410
+    dataset_root: /home/runner/work/gtsfm/gtsfm/Tanks_and_Temples_Barn_410
+    # dataset_root: /home/tdriver6/sfm_datasets/Tanks_and_Temples_Barn_410
+    scene_name: Barn
```

Comment:
Remove comments lines, also what's the role of this dataset_root variable? Is this new?

Reply:
Since the `CorrespondenceGeneratorBase` class and its child classes don't have access to the `loader`, to avoid modifying all the other APIs, for this special case we need to instantiate a loader, so we need data paths to do so (to access loader functionality like creating meshes etc.)